packages/lisk-transactions/src/utils/validation/schema.js

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+export const baseTransaction = {
+	$id: 'lisk/base-transaction',
+	type: 'object',
+	required: [
+		'id',
+		'type',
+		'amount',
+		'fee',
+		'senderPublicKey',
+		'recipientId',
+		'timestamp',
+		'asset',
+		'signature',
+	],
+	properties: {
+		id: {
+			type: 'string',
+			format: 'number',
+			minLength: 1,
+			maxLength: 20,
+		},
+		amount: {
+			type: 'string',
+			format: 'number',
+		},
+		fee: {
+			type: 'string',
+			format: 'number',
+		},
+		type: {
+			type: 'integer',
+			minimum: 0,
+			maximum: 7,
+		},
+		timestamp: {
+			type: 'integer',
+		},
+		senderId: {
+			type: 'string',
+			format: 'address',
+		},
+		senderPublicKey: {
+			type: 'string',
+			format: 'publicKey',
+		},
+		senderSecondPublicKey: {
+			type: 'string',
+			format: 'publicKey',
+		},
+		recipientId: {
+			type: 'string',
+		},
+		recipientPublicKey: {
+			type: ['string', 'null'],
+			format: 'publicKey',
+		},
+		signature: {
+			type: 'string',
+			format: 'signature',
+		},
+		signSignature: {
+			type: 'string',
+			format: 'signature',
+		},
+		signatures: {
+			type: 'array',
+			uniqueItems: true,
+			items: {
+				type: 'string',
+				format: 'signature',
+			},
+		},
+		asset: {
+			type: 'object',
```

Comment:
Should we be more specific here already?

Reply:
depending on type, it would be different, and merged so I think as `base` it should be enough